python/ray/data/impl/block_list.py

```diff
@@ -104,7 +104,6 @@ def __iter__(self):
 
             def __next__(self):
                 ref, meta = next(self._base_iter)
-                assert isinstance(ref, ray.ObjectRef), (ref, meta)
                 return ref
```

Comment:
Why remove this assert?

Reply:
This currently fails since we're shoe-horning `ReadTasks` directly into a block list, without `ray.put`ing them.